### PR TITLE
feat:billable charge list 500

### PR DESCRIPTION
**Pull request by tbo47**

### CODE REVIEW

File: src/services/QueryService.ts [hunk 1]
```diff
@@ -96,23 +95,31 @@ export class QueryService<T> {
...
-        result.meta = metadata;
-        return result;
-      });
-  }
-  async then(done: any, fail?: any): Promise<any> {
+    const [response, metadata] = await Promise.all([this.httpGet(this.parameters), this.meta.getFull(this.parameters.fields, this.parameters.layout)]);
+    const result = response.data;
+
+    // TODO look at what was asked and what is return. And ask for more
+    const needMore = (this.parameters.count - result.count > 0) && this.parameters.fields.includes('billingCalendarInstances') && this.parameters.count === 500;
```
Comment (ndickerson) [inline src/services/QueryService.ts:104]: We don't want to bake in a special case just for billingCalendarInstances. That just happens to be the only place that this currently occurs in the app, but we want this to apply anytime that more records are requested than are returned. We want to ask for more as long as there are more to ask for.  It's important to check that rest actually has more records, and can actually return them. See the java code at: https://github.com/bullhorn/dataloader/blob/master/src/main/java/com/bullhorn/dataloader/rest/RestApi.java#L240 for an example of exactly what you are wanting to do here.

File: src/services/QueryService.ts [hunk 2]
```diff
@@ -96,23 +96,57 @@ export class QueryService<T> {
...
+    if (nextStart < total && count !== 0) {
+      return nextCount;
+    }
+    return 0;
+  }
+
+  private getNext(start: number, count: number) {
+    const nextStart = start + count;
+    const alreadyFetchRecord = nextStart - this.parameters.start;
+    const nextCount = this.MAX_RECORDS - alreadyFetchRecord;
```
Comment (ndickerson) [inline src/services/QueryService.ts:141]: Super minor - consider changing variable name to: alreadyFetched

File: .gitignore [hunk 3]
```diff
@@ -37,6 +37,7 @@ lib/
 
 # Dependency directories
 node_modules/
+node_modules
```
[the remark this answers is not in the document]
Comment (tbo47) [reply .gitignore:40]: Nope `node_modules` is enough. It covers the directory `node_modules/*` and the symlink `node_modules`. I use the symlink to share the node_modules between novo and taurus.

File: src/services/QueryService.ts (hunk 2 above)
Comment (tbo47) [reply src/services/QueryService.ts]: done

**Review by ndickerson — DISMISSED** [review]

**Review by ndickerson — APPROVED** [review]

**Pull request merged**